src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -782,6 +785,90 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_inner_equi_join(
   return output_table_statistics;
 }
 
+std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_semi_join(
+    const ColumnID left_column_id, const ColumnID right_column_id, const TableStatistics& left_input_table_statistics,
+    const TableStatistics& right_input_table_statistics) {
+  // This is based on estimate_inner_equi_join. We take the histogram from the right, set the bin heights to the
```

Comment:
I am not familiar with the whole join estimation here. But I would have helped me a lot to know that inner estimation works like "use the principle of inclusion that says that for each distinct value of the smaller side we expect a match". This way, your idea becomes clearer.

Then again: do we need this creation of new histograms if the final estimation is "that simple" in the semi case?

Reply:
Ignore the last paragraph. Makes sense.

Reply:
That's documented in estimate_inner_equi_join_of_bins, which is also called by this method. I am not sure where to duplicate this comment. Any suggestion?

Reply:
I would not duplicate the comments, but just give a little bit more context in your explanation.  I think it's better to get the general idea in the first comment without stepping three methods down to find it.

Reply:
`      // If we have histograms, we use the principle of inclusion to determine the number of matches between two bins.
`?

Reply:
Seems simple, but would have helped me. :)